subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5707,6 +5797,16 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			goto error;
 		}
 
+		int session_cache = TLS_SESSION_CACHE_ENABLED;
+
+		ret = setsockopt(client_ctx->sock_fd, SOL_TLS, TLS_SESSION_CACHE, &session_cache,
```

Comment:
upstream session cache support is still on a PR (https://github.com/zephyrproject-rtos/zephyr/pull/43710)

Reply:
Interesting. It works in this repository. Re-connect just use 3 message vs 6 at full handshake.

Reply:
Ah I understand nrf91 is supporting that.